### PR TITLE
[WIP] Add wine-staging 2.0-rc1

### PR DESCRIPTION
**Pull request by yurikoles**
*If there’s a checkbox you can’t complete for any reason, that's okay, just explain in detail why you weren’t able to do so.*

After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference].
- [ ] `brew cask install {{cask_file}}` worked successfully.
- [ ] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked that the cask was not already refused in [closed issues].

I have problem with cask, install ends with message ```Error: pkg source file not found: '/usr/local/Caskroom/wine-staging/1.9.23/wine-staging--1.9.23.pkg'```

Full logs here: https://gist.github.com/yurikoles/fd1f99586a46d41d654d62510de9bb78

### CODE REVIEW

File: Casks/wine-staging.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+cask 'wine-staging' do
+  version '1.9.23'
+  sha256 '20b6f450642e3ec2ffc2a170ad5f6d8872b4389960500f1421534385c5ea2ae3'
+
+  url "https://repos.wine-staging.com/macosx/i686/winehq-staging-#{version}.pkg"
+  name 'Wine-staging'
+  homepage 'https://www.wine-staging.com/'
+
```
Comment (miccal) [inline Casks/wine-staging.rb]: Please add the line `depends_on cask: 'xquartz'` - see [here](https://github.com/wine-compholio/wine-staging/wiki/Installation#mac-os-x).

File: Casks/wine-staging.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  pkg "wine-staging--#{version}.pkg"
```
Comment (miccal) [inline Casks/wine-staging.rb]: This should be `pkg "winehq-staging-#{version}.pkg"`.

**Comment by miccal** [comment]
The `pkg` stanza requires an [`uninstall`](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/uninstall.md). Please add the following:
```
  uninstall pkgutil: [
                       'org.winehq.wine-staging',
                       'org.winehq.wine-staging-deps',
                       'org.winehq.wine-staging-deps64',
                       'org.winehq.wine-staging32',
                       'org.winehq.wine-staging64',
                     ]

```

File: Casks/wine-staging.rb (hunk 2 above)
Comment (yurikoles) [inline Casks/wine-staging.rb]: It now reports success install, but I can't run it, e.g.
```
LC_ALL=C wine
-bash: wine: command not found
``

**Comment by yurikoles** [comment]
Up & rebase.

**Comment by miccal** [comment]
Does this need [Wine](https://github.com/Homebrew/homebrew-core/blob/master/Formula/wine.rb) to be installed also? If so, you will also need to add the line `depends_on formula: 'wine'`.

**Comment by yurikoles** [comment]
Wine-Staging is a bleeding-edge test branch of wine, it contains wine binary itself.

**Comment by miccal** [comment]
Does it work if you install the `pkg` manually?

**Comment by yurikoles** [comment]
Yes, but I don't see any mention of "winehq" in `pkgutil --pkgs`.

**Comment by miccal** [comment]
But you can run it when you install it manually? Do you have Xquartz installed?

**Comment by yurikoles** [comment]
@miccal yes, but it doesn't use it.

**Comment by miccal** [comment]
I installed this via `brew cask install https://raw.githubusercontent.com/yurikoles/homebrew-cask/8f0463f8c16148e482cc04c7720f4e338d54bec8/Casks/wine-staging.rb` on a machine running macOS 10.12.1 and Xquartz 2.7.11. It created an .app called `Wine Staging.app`, which when I run opens a Terminal window with the following output:
```
################################################################################
#                           Wine Is Not an Emulator                            #
################################################################################

 Welcome to wine-1.9.23 (Staging).

 In order to start a program:
   .exe: wine program.exe
   .msi: wine msiexec /i program.msi

 If you want to configure wine:
   winecfg

 To get information about app compatibility:
   appdb Program Name

-bash-4.4.5- /Users/USER [> 
```
Running the [suggested command](https://github.com/wine-compholio/wine-staging/wiki/Installation#mac-os-x) `wine --check-libs` results in the output:
```
libcups.2.dylib: /usr/lib/libcups.2.dylib
libfreetype.6.dylib: /opt/x11/lib/libfreetype.6.dylib
libGL.1.dylib: /opt/x11/lib/libGL.1.dylib
libjpeg.62.dylib: /Applications/Wine Staging.app/Contents/Resources/wine/lib/libjpeg.62.dylib
libncurses.5.4.dylib: /usr/lib/libncurses.5.4.dylib
libnetapi.dylib: missing
libodbc.dylib: missing
libOSMesa.8.dylib: /opt/x11/lib/libOSMesa.8.dylib
libpcap.A.dylib: /usr/lib/libpcap.A.dylib
libpng15.15.dylib: /opt/x11/lib/libpng15.15.dylib
libtiff.5.dylib: /Applications/Wine Staging.app/Contents/Resources/wine/lib/libtiff.5.dylib
libtxc_dxtn.dylib: /Applications/Wine Staging.app/Contents/Resources/wine/lib/libtxc_dxtn.dylib
libX11.6.dylib: /opt/x11/lib/libX11.6.dylib
libX11-xcb.1.dylib: /opt/x11/lib/libX11-xcb.1.dylib
libXcomposite.1.dylib: /opt/x11/lib/libXcomposite.1.dylib
libXcursor.1.dylib: /opt/x11/lib/libXcursor.1.dylib
libXext.6.dylib: /opt/x11/lib/libXext.6.dylib
libXi.6.dylib: /opt/x11/lib/libXi.6.dylib
libXinerama.1.dylib: /opt/x11/lib/libXinerama.1.dylib
libXrandr.2.dylib: /opt/x11/lib/libXrandr.2.dylib
libXrender.1.dylib: /opt/x11/lib/libXrender.1.dylib
libxslt.1.dylib: /usr/lib/libxslt.1.dylib
libXxf86vm.1.dylib: /opt/x11/lib/libXxf86vm.1.dylib
```

However, if I open a new Terminal window and enter the same command I get `-bash: wine: command not found` as you did. So it seems that you MUST run the app `Wine Staging.app` and use the Terminal window that it opens.

Like yourself, I don't see any mention of "winehq" in `pkgutil --pkgs`.

In conclusion, I would say that the Cask installs successfully.

I hope all this helps.

File: Casks/wine-staging.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+  pkg "winehq-staging-#{version}.pkg"
+
+  uninstall pkgutil: [
+                       'org.winehq.wine-staging',
+                       'org.winehq.wine-staging-deps',
+                       'org.winehq.wine-staging-deps64',
+                       'org.winehq.wine-staging32',
+                       'org.winehq.wine-staging64',
+                     ]
```
Comment (miccal) [inline Casks/wine-staging.rb]: Please add `delete: '/Applications/Wine Staging.app'` to the `uninstall` stanza.

Comment (reitermarkus) [inline Casks/wine-staging.rb]: Is this not uninstalled using `pkgutil`?

Comment (miccal) [inline Casks/wine-staging.rb]: No - when I ran `brew cask uninstall https://raw.githubusercontent.com/yurikoles/homebrew-cask/8f0463f8c16148e482cc04c7720f4e338d54bec8/Casks/wine-staging.rb` it did not remove the app.

Comment (palxex) [inline Casks/wine-staging.rb]: Yeah, thats why I dont commit my wine-staging cask. Seems all those pkg dont appear in `pkgutil --pkgs|grep wine` after installing the pkg, though the pkg really include the 4 sub-pkg with the name mentioned above.

**Comment by Numbermaniac** [comment]
> However, if I open a new Terminal window and enter the same command I get -bash: wine: command not found as you did. So it seems that you MUST run the app Wine Staging.app and use the Terminal window that it opens.

This is because Wine Staging.app adds the `wine` binary to the path first when it opens Terminal. If you scroll up in the Terminal window, this is the code it runs first:

`test "$?BASH_VERSION" = "0" || eval 'setenv() { export "$1=$2"; }'; setenv PATH "/Applications/Wine Devel.app/Contents/Resources/start/bin:/Applications/Wine Devel.app/Contents/Resources/wine/bin:$PATH"; winehelp --clear`

**Comment by miccal** [comment]
Thanks for that @Numbermaniac.

As @palxex mentioned, the 5 packages do not appear in `pkgutil --pkgs` which then affects the `uninstall` as pointed out by @reitermarkus.

Ping @caskroom/maintainers - should we add this Cask?

**Comment by vitorgalvao** [comment]
I have no issue with the `app` having to be deleted separately. It seems to be fairly common in `pkg` casks, as for some reason many times they are left behind (sometimes even just an empty `app` bundle). If it leaves many other things behind, that's another matter.

[I have no strong feelings one way or the other](https://youtu.be/CxK_nA2iVXw).

**Comment by miccal** [comment]
Whoops, thanks @vitorgalvao.

**Comment by reitermarkus** [comment]
@yurikoles, can you confirm this is working now?

File: Casks/wine-staging.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+cask 'wine-staging' do
+  version '1.9.23'
```
Comment (Numbermaniac) [inline Casks/wine-staging.rb]: Since this cask was created, the latest version has now become 2.0-rc1.

**Comment by Numbermaniac** [comment]
Also, does this enable the 64-bit Wine option that is offered when installing manually?

File: Casks/wine-staging.rb (hunk 4 above)
Comment (miccal) [inline Casks/wine-staging.rb]: Assuming the `rc1` refers to Release Candidate 1, I think we will keep it at he latest "stable" version for now.

**Comment by miccal** [comment]
> Also, does this enable the 64-bit Wine option that is offered when installing manually?

@yurikoles can you confirm this? I do not have my testing machine available at the moment.

**Comment by yurikoles** [comment]
@miccal it doesn't install 64-bit version now.

**Comment by miccal** [comment]
Installing the `pkg` manually, you are presented with the following:

screen shot 2016-12-17 at 7 29 31 pm

I am not sure how to select that option via Cask.

File: Casks/wine-staging.rb (hunk 4 above)
Comment (miccal) [inline Casks/wine-staging.rb]: Updated to 2.0-rc1.

**Comment by adidalal** [comment]
@miccal https://github.com/Homebrew/brew/pull/1535
Ref: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/pkg.md

**Comment by Numbermaniac** [comment]
@adidalal Do we need to set values for the two that are already selected? Or do we just need to enable the third one only?

**Comment by miccal** [comment]
Thanks @adidalal but I was under the impression that this does not allow the user to actually make a choice?

**Comment by adidalal** [comment]
No, I don't believe so - that wasn't the intention of the feature, though, more so to allow for "optional" components to be auto-selected and installed so that we can move away from `installer :manual`

**Comment by Numbermaniac** [comment]
IMO 64-bit should just be enabled for everyone, there isn't much of a reason why you wouldn't want it.

**Comment by miccal** [comment]
Closed in favour of [#27937](https://github.com/caskroom/homebrew-cask/pull/27937).
